### DISCUSSION TITLE
Integrating Redis Object Cache into custom WordPress Plugin

### DISCUSSION THREAD

**Discussion by afzouni** [rhubarbgroup/.github#50] [Support (ROC)] [answered]
I've got a WordPress plugin I've been working on, and it's using `wpdb` to interact with a custom table in the WordPress database. Now, I'm thinking about adding Redis Object Cache to the mix to speed things up. 

Basically, I want to check if there's cached data before hitting the database, especially when I'm pulling records from my custom table (i.e. `my_table`).  Any ideas on how I can smoothly integrate this caching magic into my plugin? 

Also, I'm a bit unsure about whether to check Redis or the `wpdb` cached data. 

Any insights on that?

**Comment by tillkruss** [comment, marked as answer]
Hey!

If you use `wpdb` directly, then there is no caching. Only top-level functions `get_posts()` will use caching.

That means you can implement your own caching layer using the `wp_cache_*()` functions.

I hope that helps.

**Reply by afzouni** [reply]
Thank you, and here's another question: Is there a difference between `wp_cache_` and the Transient API, specifically `set_transient()` and `get_transient()`?

**Reply by tillkruss** [reply]
Yes, big difference. Transients will use the object cache when it's present, otherwise they will use the `options` table for caching which can be quite slow.

**Reply by afzouni** [reply]
Thank you so much
